### PR TITLE
SAK-44805 - Fixing up TaskServiceTest

### PR DESCRIPTION
**Pull request by jonespm**
@ern looks like 0170595cf4f48413922dffe26bdb04410c1bb359 broke the TaskServiceTest. 

I researched this a bit.
* Added the missing beans in the `TaskServiceTestConfiguration` that were recently added
Then it was still failing so

* I switched the assertion to use `assertEquals `to display what was actually failing and improved some of the messages.

But it was still showing the old value. I noticed there was a code change
https://github.com/sakaiproject/sakai/blob/df030150d6815c2c72a09d3f918632aa34b6fa71/kernel/kernel-impl/src/main/java/org/sakaiproject/tasks/impl/TaskServiceImpl.java#L120

And this Test had
https://github.com/sakaiproject/sakai/blob/df030150d6815c2c72a09d3f918632aa34b6fa71/kernel/kernel-impl/src/test/java/org/sakaiproject/tasks/impl/test/TaskServiceTest.java#L60

So I switched that to false since this really isn't a system test. Not sure if there was a reason "System" tasks aren't updating or if the test was just wrong to begin with. Perhaps the test could be improved to test this new System test case. 

Thanks!

### CODE REVIEW

**Review by ern — APPROVED** [review]

**Comment by jonespm** [comment]
This passed for me locally last night but was still failing on Github. I'm re-running the full build locally again.

**Comment by csev** [comment]
Please hurry and then restart all those other jobs if you can.  Thanks!

**Comment by jonespm** [comment]
Unfortunately after this is fixed, all those other PR's would have to have their branches updated to include this fix before their tests would pass.

**Comment by jonespm** [comment]
Looks like Github actions is still failing. When I run it locally I get a successful build. 

```
[INFO] Running org.sakaiproject.tasks.impl.test.TaskServiceTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.282 s - in org.sakaiproject.tasks.impl.test.TaskServiceTest
```

I'm not 100% sure this is checking out the right commmit in the Github action. I see in the step. Is this even correct? Why would it checkout the previous commit here?

```
Run git checkout HEAD^
Previous HEAD position was 3fe4ec31 Merge c2d552076673322a81ebc18d6a29c7dd424d2a43 into df030150d6815c2c72a09d3f918632aa34b6fa71
HEAD is now at df030150 SAK-44877 - Lessons: getPrerequisiteDataForSubNav NullPointerException (#8967)
```

**Review by ern — CHANGES_REQUESTED** [review]
that checkout is needed don't remove 3334f8b

**Comment by jonespm** [comment]
@ern Are you sure? The test is passing with the change I made to the maven.yml. Why was that checkout needed? From what I can see that change you'd made checks out the **previous** commit, which is doesn't seem like what we'd want from a PR checkout?

**Comment by ern** [comment]
That checkout is needed because PR's will not checkout the latest updates from the repo.

**Comment by jonespm** [comment]
Yes, that's what I added ;)

**Comment by csev** [comment]
Hey all checks have passed - lets unbreak my heart :)

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -15,10 +15,11 @@ jobs:
     runs-on: ubuntu-20.04
 
     steps:
+    # https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit
     - uses: actions/checkout@v2
       with:
         fetch-depth: 2
-    - run: git checkout HEAD^
+    - run: git checkout HEAD
```
Comment (ern) [inline .github/workflows/maven.yml:22]:
```suggestion
    - run: git checkout HEAD^
```

**Review by ern — APPROVED** [review]

**Pull request merged**